src/ood-preview/lib/ood_preview/utils.ml

```diff
@@ -0,0 +1,33 @@
+open Import
+
+let extract_metadata_body s =
+  match Str.split (Str.regexp "---") s with
+  | [ metadata; body ] ->
+    metadata, String.trim body
+  | metadata :: splitted_body ->
+    metadata, String.trim (String.concat "---" splitted_body)
+  | _ ->
+    Logs.app (fun m -> m "%s" s);
+    raise (Exn.Decode_error "expected metadata at the top of the file")
+
+let decode_or_raise f x =
+  match f x with Ok x -> x | Error (`Msg err) -> raise (Exn.Decode_error err)
+
+let read_from_dir dir =
+  let len = String.length dir in
+  Data.file_list
+  |> List.filter_map (fun x ->
+         if String.prefix x len = dir then
+           Data.read x
+         else if Glob.matches_glob ~glob:dir x then
+           Data.read x
+         else
+           None)
+
+let map_files f dir = read_from_dir dir |> List.map f
+
+let slugify value =
```

Comment:
I'm wondering how best to do relational entries in the metadata, the way it happened before through NetlifyCMS was either custom or you could just specify a field from one collection to another (e.g. the conference could reference a video by adding the video title field), but "slugifying" them might be a better option, the question is how to share this with the front-end, any thoughts?

Reply:
The slugs are only used for URLs at the moment, but if we're going to support relations in the data (which I would refrain from doing as much as possible because of the extra complexity), I think we'll need a better heuristic (e.g. the file names seem better for the data located in folders)

I use this function in `id_of_t` or `get_by_id`, which is confusing, I should replace ID by slug, which is more accurate.